src/Accounts/Accounts/Token/GetAzureRmAccessToken.cs

```diff
@@ -108,6 +112,25 @@ public override void ExecuteCmdlet()
                 UserId = accessToken.UserId,
             };
             result.ExpiresOn = (accessToken as MsalAccessToken)?.ExpiresOn ?? result.ExpiresOn;
+            if(result.ExpiresOn == default(DateTimeOffset))
```

Comment:
With your new logic, we can remove line 114
```suggestion
            if ((accessToken as MsalAccessToken)?.ExpiresOn == default(DateTimeOffset))
```